src/network/networkd-link.c

```diff
@@ -421,6 +421,14 @@ static int link_new(Manager *manager, sd_netlink_message *message, Link **ret) {
         if (r < 0)
                 log_link_debug_errno(link, r, "New device has no master, continuing without");
 
+        r = sd_netlink_message_read_data(message, IFLA_ADDRESS, &link->hw_addr_len, (void **)&link->hw_addr);
+        if (r < 0)
+                log_link_debug_errno(link, r, "Hardware address not found for new device, continuing without");
+
+        r = sd_netlink_message_read_data(message, IFLA_BROADCAST, &link->bcast_addr_len, (void **)&link->bcast_addr);
+        if (r < 0)
+                log_link_debug_errno(link, r, "Broadcast address not found for new device, continuing without");
+
         r = sd_netlink_message_read_ether_addr(message, IFLA_ADDRESS, &link->mac);
```

Comment:
Hmm, the same attribute is read to `link->mac` and `link->hw_addr`. I think these should be merged.

Reply:
There are a lot of instances in the code which expect the "struct ether_addr". I guess one could just cast the uint8_t* to the struct whereever it's expected, since the memory layout seems compatible otherwise.
But I'm not sure if such a change should be part of this PR, or a follow up?